Library/Homebrew/dependency.rb

```diff
@@ -94,11 +94,17 @@ def expand(dependent, deps = dependent.deps, cache_key: nil, &block)
       @expand_stack ||= []
       @expand_stack.push dependent.name
 
-      if cache_key.present?
-        cache[cache_key] ||= {}
-        return cache[cache_key][dependent.full_name].dup if cache[cache_key][dependent.full_name]
+      if cache_key.blank?
```

Comment:
This caching logic is starting to look pretty involved and making me uneasy. I'd really love if we can use both a `extend Cachable` and try to push these to a global switch (even if it requires a wider refactoring).

Reply:
We already `extend Cachable`.

It's possible to have something global for caching that lasts one expansion, but it's not possible to do that and cache things over multiple calls (since the input block may be different).

Reply:
> We already `extend Cachable`.

🤦🏻 sorry.

> It's possible to have something global for caching that lasts one expansion, but it's not possible to do that and cache things over multiple calls (since the input block may be different).

Yeh, I'm wondering if we could/should move away from this block expansion approach.

Reply:
I'm not sure what the alternative would look like at the moment so I'll need to think it through a bit.